### PR TITLE
feat: for

### LINKED ISSUE #138: for executes its body only once
The following program:

```
  let i
  for(  i = 0; i < 5; i = i+1) {
    if (-1 < i) {
      log(i)
    }
  }
```
run in AskQL cli outputs:

```
0
any ask(let('i'),for(list(let('i',0)),list(call(get('<'),get('i'),5)),list(let('i',call(get('+'),get('i'),1))),block(list(let('i',0)),if(call(get('<'),-1,get('i')),block(call(get('log'),get('i'))),block()),list(let('i',call(get('+'),get('i'),1))))))
[ 1 ]
```

Instead of just `0`, 
```
0
1
2
3
4
```
was expected.

### PR DESCRIPTION
**Pull request by mhagmajer**
Close #138

### CODE REVIEW

File: src/askscript/__tests__/99-e2e-tests/e2e-12-bubble_sort.ask [hunk 1]
```diff
@@ -0,0 +1,13 @@
+// Bubble sort
+ask {
+  let arr = [342.34, 35767, 2, 117, 999, 3435]
+  const len = arr:length
+  for (let i = 0; i < len; i = i + 1) {
+	  for (let j = 0; (j + 1) < len; j = j + 1){
```
Comment (mhagmajer) [inline src/askscript/__tests__/99-e2e-tests/e2e-12-bubble_sort.ask:6]: @czerwinskilukasz1 is this right or should it say `j < len`?

Comment (km4) [reply src/askscript/__tests__/99-e2e-tests/e2e-12-bubble_sort.ask:6]: len its fine but problem is in the const `const len = arr:length` :) should be 'dot' `arr.length`

File: src/askjsx/lib/For.tsx [hunk 2]
```diff
@@ -4,7 +4,7 @@ askjsx;
 
 export function For({
   initialization,
-  condition,
+  condition = true,
```
Comment (km4) [inline src/askjsx/lib/For.tsx:7]: @czerwinskilukasz1 
wow First time see on object assignment `=` its correct or should be `:` ?

File: src/askscript/__tests__/99-e2e-tests/e2e-12-bubble_sort.ask (hunk 1 above)
Comment (mhagmajer) [reply src/askscript/__tests__/99-e2e-tests/e2e-12-bubble_sort.ask:6]: This is AskScript not JavaScript :)
`arr` can be called with `length` so `arr:length`
it's the same as `length(arr)`

File: src/askscript/__tests__/99-e2e-tests/e2e-12-bubble_sort.ask [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+		  if (arr:at(j) > arr:at(j + 1)) {
+			  arr = arr:set(j, arr:at(j + 1)):set(j + 1, arr:at(j))
```
Comment (mhagmajer) [inline src/askscript/__tests__/99-e2e-tests/e2e-12-bubble_sort.ask:8]: @czerwinskilukasz1 look how this expression is evaluated by runtime 🙀. this is `swap(at[j], at[j+1]`

File: src/askjsx/lib/For.tsx (hunk 2 above)
Comment (mhagmajer) [reply src/askjsx/lib/For.tsx:7]: ES2015

File: src/askscript/__tests__/99-e2e-tests/e2e-12-bubble_sort.ask (hunk 3 above)
Comment (czerwinskilukasz1) [reply src/askscript/__tests__/99-e2e-tests/e2e-12-bubble_sort.ask:8]: @mhagmajer , cool.  
By the way, how about we implement a swap resource? :) 
https://github.com/xFAANG/askql/issues/213

File: src/askscript/__tests__/99-e2e-tests/e2e-12-bubble_sort.ask (hunk 1 above)
Comment (czerwinskilukasz1) [reply src/askscript/__tests__/99-e2e-tests/e2e-12-bubble_sort.ask:6]: I would 

> @czerwinskilukasz1 is this right or should it say `j < len`?

I think it should be rather `j < len-i-1`, following https://www.geeksforgeeks.org/bubble-sort/
Of course `j < len` should work too.

File: src/askscript/__tests__/99-e2e-tests/e2e-12-bubble_sort.ask (hunk 3 above)
Comment (mhagmajer) [reply src/askscript/__tests__/99-e2e-tests/e2e-12-bubble_sort.ask:8]: @czerwinskilukasz1 I guess what I meant was to show you that we actually have a working swap inliner in this language :)

File: src/askscript/__tests__/99-e2e-tests/e2e-12-bubble_sort.ask (hunk 1 above)
Comment (mhagmajer) [reply src/askscript/__tests__/99-e2e-tests/e2e-12-bubble_sort.ask]: @czerwinskilukasz1 done

**Comment by czerwinskilukasz1** [comment]
LGTM.  @km4, any additional comments ?

**Comment by czerwinskilukasz1** [comment]
(interesting, the button for Re-requesting km4's review doesn't react, let's wait for a bit for his reply)

**Pull request merged**